dsi/offline/acceptance/experiment.py

```diff
@@ -0,0 +1,114 @@
+        super().__init__(config, gen_config)
+        self.draft_gen_config: ConfigGen = draft_gen_config
+
+    def _load_draft_model_tokenizer(self) -> tuple:
+        log.info(
+            f"Loading model: {self.config.draft_model}, \
+            compile={self.config.draft_compile_model}"
+        )
+        model = AutoModelForCausalLM.from_pretrained(
+            self.config.draft_model,
+            device_map="auto",
+            torch_dtype=self.config.get_torch_draft_dtype(),
+            revision=self.config.draft_revision,
+        )
+        tokenizer = AutoTokenizer.from_pretrained(self.config.draft_model)
+        model = torch.compile(model) if self.config.draft_compile_model else model
+        return model, tokenizer
+
+    def _are_tokenizers_same(self, tokenizer1, tokenizer2) -> bool:
+        # Compare vocabularies
+        if tokenizer1.get_vocab() != tokenizer2.get_vocab():
+            return False
+
+        # Compare special tokens
+        special_tokens_to_compare = [
+            "eos_token_id",
+            "pad_token_id",
+            "bos_token_id",
+            "unk_token_id",
+        ]
+        for token in special_tokens_to_compare:
+            if getattr(tokenizer1, token, None) != getattr(tokenizer2, token, None):
+                return False
+
+        return True
+
+    def _single_repeat(self) -> ResultAcceptance:
+        all_n_matches = []
+
+        examples = self._get_random_prompted_examples()
+
+        target_model, target_tokenizer = self._load_model_tokenizer()
+        draft_model, draft_tokenizer = self._load_draft_model_tokenizer()
+
+        # Check if tokenizers are the same
+        if not self._are_tokenizers_same(target_tokenizer, draft_tokenizer):
+            raise ValueError("The target and draft tokenizers are not the same.")
+
+        target_gen_kwargs = dict(
+            do_sample=self.gen_config.do_sample,
+            temperature=self.gen_config.temperature,
+            top_p=self.gen_config.top_p,
+            pad_token_id=target_tokenizer.eos_token_id,
+            max_new_tokens=self.config.max_new_tokens,
+        )
+
+        draft_gen_kwargs = dict(
+            do_sample=self.draft_gen_config.do_sample,
+            temperature=self.draft_gen_config.temperature,
+            top_p=self.draft_gen_config.top_p,
+            pad_token_id=target_tokenizer.eos_token_id,
+            max_new_tokens=1,
+        )
+
+        for ex in tqdm(examples):
+            inputs = target_tokenizer(ex, return_tensors="pt").to(target_model.device)
+            n_matches = [0]
+            output_target = target_model.generate(**inputs, **target_gen_kwargs)
+            prompt_len = len(inputs.input_ids[0])
+
+            # iterate over the tokens generated by the target and
+            # check whether the draft produces the same token
+            for i in range(prompt_len, len(output_target[0])):
+                inputs["input_ids"] = output_target[0, 0:i].view(1, -1)
```

Comment:
It's not clear what `output_target` refers to and why we access it with `output_target[0, 0:i]`. To enhance readability and make the code easier to understand without needing to run debug mode or insert print statements, consider encapsulating this logic in a descriptive function. Here's a proposed function:
```python
def get_input_ids_prefix(output_seqs, tok_pos_last: int):
    """Returns a prefix of a sequence of input ids.
       Args: 
        output_seqs: The output from Hugging Face's transformers `model.generate` function, expected to be a sequence-like data structure.
        tok_pos_last: int - The last token position to include in the returned sequence.
    """
    return output_seqs[0, 0:tok_pos_last]
```